src/main/scala/com/mozilla/telemetry/sinks/HTTPSink.scala

```diff
@@ -17,8 +33,32 @@ class HttpSink[String](url: String, parameters: Map[String, String]) extends For
   }
 
   def process(event: String): Unit = {
-    getRequest
-      .param("event", event.toString)
-      .asString
+    attempt(
+      getRequest
+        .param("event", event.toString)
+        .timeout(connTimeoutMs = connectionTimeout, readTimeoutMs = ReadTimeout))
+  }
+
+  private def backoff(tries: Int): Long = (scala.math.pow(2, tries) - 1).toLong * defaultDelay
+
+  @tailrec
+  private def attempt(request: HttpRequest, tries: Int = 0): Unit = {
+    if(tries > 0) { // minor optimization
+      java.lang.Thread.sleep(backoff(tries))
+    }
+
+    val code = Try(request.asString.code) match {
+      case Success(c) => c
+      case Failure(e: java.net.SocketTimeoutException) => TimeoutPseudoCode
```

Comment:
Can we get any other exception here?
If so, we won't send events remaining in current batch.

Reply:
Added catching and logging any `NonFatal` exceptions. I think we want fatal ones to follow through, since it will cause the job to fail (assuming the error reproduces on retries of the partition), in which case we can investigate.

Reply:
You're right, I didn't think of `Fatal` ones.